### ISSUE TITLE
Do we need FunctionContext like DecoratorContext?

### ISSUE THREAD

**Issue by nguerrera** [microsoft/typespec#523]
When a projection calls a function, it passes a bare `Program` as the first arg automatically. Will we eventually need more info for the same reasons we added DecoratorContext? Or could this be a single `Context` type?

**Comment by bterlson** [comment]
I would like to see these be the same, personally.

**Comment by timotheeguerin** [comment]
Yeah could probably be the exact same type

**Comment by nguerrera** [comment]
Maybe we should rename the current type ASAP then to avoid delaying a (TS source) break too long.

**Comment by timotheeguerin** [comment]
Yeah probably be good but we could also have that later and have a smooth transition if there is a reason to wait 
```ts
/**
 *  @deprecated use FunctionContext
 */
type DecoratorContext = FunctionContext;
```

**Comment by markcowl** [comment]
Is there aany design to discuss here, or can we just go to implementation, with the alias for DecoratorContext?

**Comment by nguerrera** [comment]
I think the only design to discuss is the name of the unified type. I think maybe `CallContext` is better than FunctionContext but I don't have a strong opinion.

**Comment by timotheeguerin** [comment]
Taken care of in function revival #9060
